src/components/task-plan/plan-mixin.coffee

```diff
@@ -39,8 +39,13 @@ module.exports =
 
   publish: ->
     {id} = @props
-    TaskPlanStore.addChangeListener(@saved)
-    TaskPlanActions.save(id)
+    publishable = TaskPlanStore.isValid(id)
+
+    if (publishable)
+      TaskPlanStore.addChangeListener(@saved)
+      TaskPlanActions.save(id)
+    else
+      @setState({invalid: true})
```

Comment:
I think this is the point that implements the "Publish button is always enabled. Show the required fields when teacher tries to publish" piece. Can you add a comment here blaming fred for why this is done this way? ; )